### PR TITLE
Add standalone inference script with zero project dependencies and modern Python/PyTorch compatibility

### PR DESCRIPTION
**Pull request by devymex**
## Summary

The original codebase depends on maskrcnn_benchmark (which includes C++ CUDA extensions and requires `python setup.py develop`), a [forked `transformers`](https://github.com/wuyongjianCODE/VLPMNuD), GLIP, YOLOX, and many other libraries with strict version constraints (e.g. `torch==1.9.0+cu111`, `Python 3.7`). This makes it very difficult to set up the environment, especially on newer systems with updated CUDA drivers and Python versions.

This PR adds a fully **self-contained inference module** under standalone that:

- **Has zero dependency on project source code** — does not import anything from maskrcnn_benchmark, GLIP, or any forked library.
- **Re-implements all necessary model components from scratch** (Swin-L backbone, FPN, BERT language backbone, VLDyHead with deformable convolutions, bi-directional VL attention, ATSS post-processing, ConvMapModule, MSTB injection) in pure PyTorch, faithfully matching the original checkpoint's parameter names and architecture.
- **Works with modern Python and library versions** — tested with Python 3.11, PyTorch 2.x, Transformers 4.x, and Torchvision 0.x (see requirements.txt).
- **Eliminates the need for C++ extension compilation** (`python setup.py develop` is no longer required for inference).
- **Loads the same pre-trained checkpoint** (`model_best.pth`) produced by the original training pipeline, with no conversion needed.

## New Files

| File | Description |
|---|---|
| infer.py | CLI entry point — handles image preprocessing, tokenization, model loading, MSTB visual-token injection, inference, and result visualization |
| modules.py | All model building blocks: `SwinTransformer`, `FPN`, `Backbone`, `LanguageBackbone` (BERT), `VLDyHead`, `BiMultiHeadAttention`, `VLFuse`, `DyConv`, `AnchorGenerator`, `ATSSPostProcessor`, `ConvMapModule`, `BoxList`, etc. |
| requirements.txt | Minimal dependencies: `torch>=1.13`, `torchvision>=0.9`, `transformers>=4.18`, `Pillow>=9.1`, `numpy>=1.20` |
| images | Sample reference and query images for quick testing |

## Usage

```bash
pip install -r standalone/requirements.txt

python standalone/infer.py \
    --ref standalone/assets/images/ref1.png \
    --img standalone/assets/images/img1.png \
    --text spoon \
    --weight /path/to/model_best.pth
```

Results are saved to standalone_results by default.

## Motivation

- **Ease of use**: Users who only need inference no longer have to build the full GLIP/maskrcnn environment.
- **Portability**: Runs on modern hardware setups and Python 3.8–3.12+ without version conflicts.
- **Reproducibility**: Lowers the barrier for verifying published results.

## Notes

- This PR only adds inference functionality. Training still requires the original environment.
- The standalone code loads the exact same `model_best.pth` checkpoint — no weight conversion is needed.
- The original codebase is untouched; this is purely additive.

## Summary by Sourcery

Add a self-contained standalone inference module for VisTex-OVLM that can run with modern PyTorch/Transformers without depending on the original project codebase or C++ extensions.

New Features:
- Introduce a standalone Swin+FPN visual backbone, BERT-based language backbone, VL fusion head, and postprocessing modules implemented purely in PyTorch for inference.
- Add a command-line inference script that loads the original VisTex checkpoint, preprocesses reference and query images, injects visual tokens, runs detection, and saves visualized results.
- Provide a minimal requirements file specifying only the modern PyTorch, Torchvision, Transformers, Pillow, and NumPy dependencies needed for standalone inference.

Enhancements:
- Implement internal utilities for box handling, anchor generation, and ATSS-style postprocessing to mirror the original model behaviour without importing project code.
- Add a ConvMapModule to map multi-scale FPN features into a single 768-dimensional visual embedding used for MSTB-style token injection.

### CODE REVIEW

File: standalone/modules.py [hunk 1]
```diff
@@ -0,0 +1,995 @@
...
+        self.bias_lang = nn.Parameter(torch.zeros(768))
+        self.bias0 = nn.Parameter(torch.tensor([-math.log((1 - 0.01) / 0.01)]))
+        self.scales = nn.ModuleList([Scale(1.0) for _ in range(5)])
+
+    def forward(self, features, language_dict_features, embedding):
+        feat_inputs = {"visual": features, "lang": language_dict_features}
+        dyhead_out = self.dyhead_tower(feat_inputs)
+
+        # Use fused hidden for embedding (USE_FUSED_FEATURES_DOT_PRODUCT=True)
+        embedding = dyhead_out["lang"]["hidden"]
```
[the remark this answers is not in the document]
Comment (devymex) [reply standalone/modules.py]: Thanks for the thorough review! All four points addressed in the latest commit:

- **`VLDyHead.forward` embedding param**: Removed it from the signature entirely — fused features are always required for this checkpoint, so there's no use case for a caller-supplied value. Updated the call site accordingly.
- **`ConvMapModule` batch size assumption**: Added an explicit `assert features[0].shape[0] == 1` with a clear error message to surface this assumption immediately rather than silently producing wrong results.
- **`CellAnchors` zero buffers**: Intentional — they are placeholders overwritten by checkpoint values via `load_state_dict`. Added a docstring to make this explicit.
- **`mstb_inject` missing label guard**: Added a `KeyError` with an informative message that includes the available label keys, so failures from unexpected captions are easy to diagnose.

**Review by WitGotFlg — APPROVED** [review]
Thank you @devymex for your excellent work
